python/arcticdb/arctic.py

```diff
@@ -40,7 +40,7 @@ class Arctic:
     _created_lib_names: Optional[List[str]] = None
     _accessed_libs: Optional[List[NativeVersionStore]] = None
 
-    def __init__(self, uri: str, encoding_version: EncodingVersion = DEFAULT_ENCODING_VERSION):
+    def __init__(self, uri: str, encoding_version: EncodingVersion = DEFAULT_ENCODING_VERSION, credential: AzureDefaultCredential = None):
```

Comment:
Throw if `credential` set but storage is not `Azure`? Throw if `credential` set and storage is `Azure` but SAS token is provided? etc